x-pack/plugins/endpoint/common/types.ts

```diff
@@ -315,23 +315,34 @@ export interface EndpointEvent {
     category: string;
     type: string;
     id: string;
+    kind: string;
   };
   endpoint: {
     process: {
       entity_id: string;
-      parent: {
+      parent?: {
         entity_id: string;
       };
+      name: string;
     };
   };
   agent: {
     id: string;
     type: string;
   };
+  process: {
+    name: string;
+  };
 }
 
 export type ResolverEvent = EndpointEvent | LegacyEndpointEvent;
 
+export function isLegacyEvent(
```

Comment:
Would you plz move this out of `types`

Reply:
❔ If its sole purpose seems to be as a type discriminant, where should it go? A selector?

Reply:
i think 'model' would be good. I'd say 'no' to selector as it doesn't take global state as its only param (doesn't take it at all).